messenger_test.go

```diff
@@ -347,9 +347,184 @@ func (s *MessengerSuite) TestChatPersistencePrivateGroupChat() {
+		Color:                 "#fffff",
+		Active:                true,
+		ChatType:              ChatTypePublic,
+		Timestamp:             3,
+		LastClockValue:        20,
+		DeletedAtClockValue:   30,
+		UnviewedMessagesCount: 40,
+	}
+
+	s.Require().NoError(s.m.SaveChat(chat1))
+	s.Require().NoError(s.m.SaveChat(chat2))
+	s.Require().NoError(s.m.SaveChat(chat3))
+
+	s.Require().NoError(s.m.SaveContact(contact))
+
+	contact.Name = "blocked"
+
+	messages := []*Message{
+		&Message{
+			ID:          "test-1",
+			ChatID:      chat2.ID,
+			ContentType: "content-type-1",
+			Content:     "test-1",
+			ClockValue:  1,
+			From:        contact.ID,
+		},
+		&Message{
+			ID:          "test-2",
+			ChatID:      chat2.ID,
+			ContentType: "content-type-2",
+			Content:     "test-2",
+			ClockValue:  2,
+			From:        contact.ID,
+		},
+		&Message{
+			ID:          "test-3",
+			ChatID:      chat2.ID,
+			ContentType: "content-type-3",
+			Content:     "test-3",
+			ClockValue:  3,
+			Seen:        false,
+			From:        []byte("test"),
+		},
+		&Message{
+			ID: "test-4",
+
+			ChatID:      chat2.ID,
+			ContentType: "content-type-4",
+			Content:     "test-4",
+			ClockValue:  4,
+			Seen:        false,
+			From:        []byte("test"),
+		},
+		&Message{
+			ID:          "test-5",
+			ChatID:      chat2.ID,
+			ContentType: "content-type-5",
+			Content:     "test-5",
+			ClockValue:  5,
+			Seen:        true,
+			From:        []byte("test"),
+		},
+		&Message{
+			ID:          "test-6",
+			ChatID:      chat3.ID,
+			ContentType: "content-type-6",
+			Content:     "test-6",
+			ClockValue:  6,
+			Seen:        false,
+			From:        contact.ID,
+		},
+		&Message{
+			ID:          "test-7",
+			ChatID:      chat3.ID,
+			ContentType: "content-type-7",
+			Content:     "test-7",
+			ClockValue:  7,
+			Seen:        false,
+			From:        []byte("test"),
+		},
+	}
+
+	err = s.m.SaveMessages(messages)
+	s.Require().NoError(err)
+
+	response, err := s.m.BlockContact(contact)
+	s.Require().NoError(err)
+
+	// The new unviewed count is updated
+	s.Require().Equal(uint(1), response[0].UnviewedMessagesCount)
+	s.Require().Equal(uint(2), response[1].UnviewedMessagesCount)
+
+	// The new message content is updated
+	s.Require().Equal("test-7", response[0].LastMessageContent)
+	s.Require().Equal("test-5", response[1].LastMessageContent)
+
+	// The new message content-type is updated
+	s.Require().Equal("content-type-7", response[0].LastMessageContentType)
+	s.Require().Equal("content-type-5", response[1].LastMessageContentType)
+
+	// The contact is updated
+	savedContacts, err := s.m.Contacts()
+	s.Require().NoError(err)
+	s.Require().Equal(1, len(savedContacts))
+	s.Require().Equal("blocked", savedContacts[0].Name)
+
+	// The chat is deleted
+	actualChats, err := s.m.Chats(0, -1)
+	s.Require().NoError(err)
+	s.Require().Equal(2, len(actualChats))
+
+	// The messages have been deleted
+	chat1Messages, _, err := s.m.MessageByChatID(chat2.ID, "", 20)
```

Comment:
The returned variable is called `chat1Messages` while we're asking for messages from `chat2`